### PR TITLE
Simulate shuffling token range endpoints

### PR DESCRIPTION
**Pull request by schlosna**
**Goals (and why)**:
Additional unit tests on top of https://github.com/palantir/atlasdb/pull/5969

==COMMIT_MSG==
Simulate shuffling token range endpoints
==COMMIT_MSG==

**Implementation Description (bullets)**:

**Testing (What was existing testing like?  What have you done to improve it?)**:

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolution.java
Comment (sudiksha27) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolution.java:59]: nit: is the explicit use of immutableSet on purpose?

File: atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolutionTest.java
Comment (sudiksha27) [inline atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolutionTest.java:146]: nit: think it will be okay to use the interface here instead of explicit immutable

**Review by sudiksha27 — APPROVED** [review]
Thanks for the fix + tests 🎖️ 
Will wait for @jeremyk-91 for merge but lgtm!

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolution.java
Comment (schlosna) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolution.java:59]: yes this is explicit to be specific that this is immutable and avoid an extra collection copy in `ImmutableIdentityAgnosticRanges` construction since we're now collecting the token range views to `ImmutableSet`s and passing those to  `ImmutableIdentityAgnosticRanges::of`
https://github.com/palantir/atlasdb/blob/8b4ff2da6f748826589c2e24bdb9014cd76fc0fa/atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolution.java#L42-L43

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolution.java
Comment (jeremyk-91) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TokenRangeResolution.java:45]: nice!

**Review by jeremyk-91 — APPROVED** [review]
nice, thanks!

**Pull request merged**
